stan/math/prim/fun/mdivide_left_tri.hpp

```diff
@@ -77,23 +74,20 @@ inline Eigen::Matrix<T, R1, C1> mdivide_left_tri(
  *
  * @tparam TriView Specifies whether A is upper (Eigen::Upper)
  * or lower triangular (Eigen::Lower).
- * @tparam T1 type of elements in the triangular matrix
- * @tparam T2 type of elements in the right-hand side matrix or vector
- * @tparam R1 number of rows in the triangular matrix, can be Eigen::Dynamic
- * @tparam C1 number of columns in the triangular matrix, can be Eigen::Dynamic
- * @tparam R2 number of rows in the right-hand side matrix, can be
- *         Eigen::Dynamic
+ * @tparam T1 type of the triangular matrix
+ * @tparam T2 type of the right-hand side matrix or vector
  *
  * @param A Triangular matrix.
  * @param b Right hand side matrix or vector.
  * @return x = A^-1 b, solution of the linear system.
  * @throws std::domain_error if A is not square or the rows of b don't
  * match the size of A.
  */
-template <Eigen::UpLoType TriView, int R1, int C1, int R2, int C2>
-inline Eigen::Matrix<double, R1, C2> mdivide_left_tri(
-    const Eigen::Matrix<double, R1, C1> &A,
-    const Eigen::Matrix<double, R2, C2> &b) {
+template <Eigen::UpLoType TriView, typename T1, typename T2,
+          require_all_eigen_t<T1, T2> * = nullptr,
+          require_all_same_vt<double, T1, T2> * = nullptr>
```

Comment:
Probably simpler to use ```require_all_eigen_vt``` here

Reply:
It would be if we had a `is_double` type trait. But we don't.

Reply:
Most OpenCL implementations only work with double. Passing in int would fail to compile.